package proto;

message Equipment {

	required string name = 1;
	required string date = 2;
	required int32 id = 3;

	message Treadmill {

		required float distance = 1;
		required float avgSpeed = 2;
	}

	message RowingMachine {

		required float duration = 1;
		required float repsPerMin = 2;
	}

	oneof EquipmentType {
		Treadmill treadmill = 4;
		RowingMachine rowingMachine = 5;
	}
}

message EquipmentCache {
	repeated Equipment equipment = 1;
}
